pallets/pool-system/src/pool_types.rs

```diff
@@ -300,16 +303,12 @@ impl<
 		> = Vec::new();
 
 		for tranche in self.tranches.residual_top_slice().iter() {
-			let metadata = AssetRegistry::metadata(&self.currency.into())
-				.ok_or(AssetMetadata {
-					decimals: 0,
-					name: Vec::new(),
-					symbol: Vec::new(),
-					existential_deposit: (),
-					location: None,
-					additional: (),
-				})
-				.unwrap();
+			let metadata = AssetRegistry::metadata(
+				&<AssetRegistry as orml_traits::asset_registry::Inspect>::AssetId::from(
+					tranche.currency.into(),
+				),
+			)
+			.ok_or(DispatchError::CannotLookup)?;
```

Comment:
Thanks for cleaning that weird `ok_or` and removing the `unwrap()`

Reply:
Yeah that code path gave me shivers